posthog/warehouse/api/saved_query.py

```diff
@@ -330,6 +349,16 @@ def update(self, instance: Any, validated_data: Any) -> Any:
                 unpause_saved_query_schedule(str(instance.id))
             sync_saved_query_workflow(view, create=not schedule_exists)
 
+        # Handle snapshot_enabled changes
+        if before_update and before_update.snapshot_enabled != view.snapshot_enabled:
+            schedule_exists = snapshot_workflow_exists(str(view.id) + "-snapshot")
+            if view.snapshot_enabled:
+                DataWarehouseSnapshotConfig.objects.get_or_create(team=view.team, saved_query=view)
+                sync_saved_query_snapshot_workflow(view, create=not schedule_exists)
+            else:
+                if schedule_exists:
+                    pause_snapshot_schedule(str(view.id) + "-snapshot")
```

Comment:
We haven't added any suffix to the schedule IDs before (not opposed to it at all) - but maybe add the suffix in the `snapshot_service.py` file to all the methods instead of relying on the top level caller to remember to add `-snapshot` to the `id`